tools/cli/src/main/java/org/apache/carbondata/tool/DataSummary.java

```diff
@@ -311,26 +309,39 @@ private void printColumnStats(String columnName) throws IOException, MemoryExcep
             max = new String(blockletMax, Charset.forName(DEFAULT_CHARSET));
           }
         } else {
-          minPercent = String.format("%.1f", blocklet.getColumnChunk().getMinPercentage() * 100);
-          maxPercent = String.format("%.1f", blocklet.getColumnChunk().getMaxPercentage() * 100);
+          // for complex columns min and max and percentage
+          if (blocklet.getColumnChunk().column.getColumnName().contains(".val") ||
+              blocklet.getColumnChunk().column.getColumnName().contains(".")) {
+            minPercent = "NA";
+            maxPercent = "NA";
+          } else {
+            minPercent =
+                String.format("%.1f", Math.abs(blocklet.getColumnChunk().getMinPercentage() * 100));
+            maxPercent =
+                String.format("%.1f", Math.abs(blocklet.getColumnChunk().getMaxPercentage() * 100));
+          }
           DataFile.ColumnChunk columnChunk = blocklet.columnChunk;
-          if (columnChunk.column.isDimensionColumn() && DataTypeUtil
+          // need to consider no  dictionary complex column
+          if (columnChunk.column.hasEncoding(Encoding.DICTIONARY) || blocklet
+              .getColumnChunk().column.getColumnName().contains(".val") || blocklet
```

Comment:
Can you add a function in ColumnSchema to return whether it is a complex column encoded in global dictionary, instead of hard coding it here

Reply:
can have a method which tells the column is complex column based on name, we already have method which tells the column by datatype, and for dictionary include and complex type, togetther no need to check, because again for child columns need to have other method, as we cant give child columns in dictionary

Reply:
handled, please review